apps/sms/test/marionette/mocks/mock_message_thread.js

```diff
@@ -0,0 +1,38 @@
+module.exports = {
```

Comment:
Regarding to injects, I just don't feel that we need such abstraction, I'd rather have it in test suite like, maybe I'm just biased and used to see what dependencies particular test requires, it's not really tied to any view:
```
const DEPENDENCIES = [ 
  '/mocks/mock_test_storages.js',
  '/mocks/mock_navigator_moz_icc_manager.js',
  '/mocks/mock_navigator_moz_mobile_message.js'
];

DEPENDENCIES.forEach((dependency) => client.contentScript.inject(__dirname + dependency));
```

Regarding to thread generation - I'm fine if we have some smart auto-generation class/object, but it's worth to generalize only when it's reusable across suites. So if you think we need it, let's make it more parametrized - at least we need MMS and different participant numbers/emails.

Also sorry for the confusion with  ```uniqueIdCounter```, I know it could be done better, but haven't had time to improve it yet. It's used as unique identifier for _all_ messages inside mock, so that newly created messages (via mozMobileMessage.send for example) won't conflict with messages inserted via mock. So this number can't be tied to number of messages, especially when you need to create several threads. We can "sit together" and think how to make it easier for test-writer :)